lib/eslint.js

```diff
@@ -807,20 +811,18 @@ module.exports = (function() {
              */
             controller.traverse(ast, {
                 enter: function(node, parent) {
-
                     var comments = api.getComments(node);
 
                     emitCommentsEnter(comments.leading);
                     node.parent = parent;
-                    api.emit(node.type, node);
+                    codePathAnalyzer.enterNode(node);
```

Comment:
Hmm, that's not very pretty. :)

I'm still concerned that we different parts emitting for different things. Can the code path analyzer just emit events for code paths and leave emitting node events in eslint?